jupyterhub/tests/selenium/test_browser.py

```diff
@@ -921,8 +922,15 @@ async def test_oauth_page(
     service_url = url_path_join(public_url(app, service) + 'owhoami/?arg=x')
     await in_thread(browser.get, (service_url))
     expected_client_id = service.name
-    expected_redirect_url = app.base_url + f"servises/{service.name}/oauth_callback"
-    assert expected_client_id, expected_redirect_url in browser.current_url
+    expected_redirect_url = url_path_join(
+        app.base_url + f"services/{service.name}/oauth_callback"
+    )
+    # decode the URL
+    decoded_browser_url = urllib.parse.unquote(
+        urllib.parse.unquote(browser.current_url)
+    )
+    assert f"client_id=service-{expected_client_id}" in decoded_browser_url
```

Comment:
Since we're asserting the values of url params, let's try parsing the URL using [urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) and then [parse_qs](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs) on the query, rather than running unquote twice on the full URL.

Reply:
thank you, done